src/RWS.jl

```diff
@@ -229,13 +229,12 @@ function BuildRWS(G::Groups.FPGroup;
     =#
 
     # Generators
-    C = [""]
-    append!(C, string.(G.gens))
-    GC.@preserve C true
+    C = Symbol[Symbol(""); getfield.(G.gens, :id)]
     # Get the number of generators
-    rws.num_gens = size(C)[1] - 1
+    rws.num_gens = length(C)
     # Move the generators to the RWS. Warning. These are not processed and are assumed to be correct.
-    rws.gen_name = Base.unsafe_convert(Ptr{Ptr{Int8}}, Base.cconvert(Ptr{Ptr{Int8}}, C))
+    ref_C = Base.cconvert(Ptr{Ptr{Int8}}, C)
+    rws.gen_name = Base.unsafe_convert(Ptr{Ptr{Int8}}, ref_C)
 
-    return rws
+    return rws, ref_C
```

Comment:
Looks better, but the following
```julia
G = let F = FreeGroup(["a", "b", "c"])
    a,b,c = Groups.gens(F)
    rels = [a^7, b^3, c*a*b]
    F/rels
end

let (rws, stuff_to_keep_alive) = KBmag.BuildRWS(G)
    KBmag.gen_names(rws)
end
```
still does not work. Any ideas?

Reply:
`rws.num_gens = length(C)` shoud be replaced by `rws.num_gens = length(C)-1`

Reply:
Okay, that was a stupid reason.

Reply:
btw. have a look at the other pull, it might be better to rebase this on the other which does `unsafe_string` instead of hand-manipulation of pointers, and so on